src/vcf/assembly_checker.cpp

```diff
@@ -121,7 +121,7 @@ namespace ebi
           if (rc.get()) {
```

Comment:
taking another look at this, why does the user (assembly_checker.cpp) need to know the IFasta implementation? if the main created one or other implementation based on the parameters, the implementations should act differently so that there shouldn't be checks here on `fetch_contig_on_demand`, right?

Reply:
Yes, this was ugly. Partly was due to previous implementation which requires passing a boolean as an argument in long chain of function calls. This has now been refactored in d599e3b. Hope this looks cleaner.